field/src/field.rs

```diff
@@ -48,6 +50,9 @@ pub trait FieldAlgebra:
 {
     type F: Field;
 
+    /// The field `ℤ/p` where the characteristic of this given ring is p.
+    type Char: PrimeField;
```

Comment:
Would this always match `F::Char`? Just wondering if it's in here for convenience or a different reason.

Reply:
Yeah this should always match `F::Char`. It's currently here because I'm going to split `FieldAlgebra` up into `FieldAlgebra` (which handles the algebra over a field structure) and `PrimeCharacteristicRing` (which handles all the underlying ring structure) in a future PR. Once that happens, `F` and `Char` won't be in the same trait any more and this should make a little more sense. (I agree that currently we shouldn't need `F::Char`).